### ISSUE TITLE
Allow backend to startup even if kubernetes config is mising

### ISSUE THREAD

**Issue by devth** [backstage/backstage#9418]
## Feature Suggestion

Currently if the kubernetes integration is installed, it will prevent the backend from starting up if the config is missing, with a log message like:

```
[1] 2022-02-09T00:20:31.214Z kubernetes info Initializing Kubernetes backend type=plugin
[1] 2022-02-09T00:20:31.214Z kubernetes info action=LoadingCustomResources numOfCustomResources=0 type=plugin
[1] Backend failed to start up, Error: Missing required config value at 'kubernetes.clusterLocatorMethods'
```

If the config is missing, could the plugin fail gracefully and allow the rest of the backend to start up?

## Possible Implementation

I'm  not sure, but I'd be willing to investigate how to do this and contribute, if people think it's something we should do.

## Context

In local development the kubernetes plugin is often not needed, and many devs don't even have access to a kubernetes cluster. In this case it'd be useful to let them continue to dev on other aspects of Backstage without needing k8s config.

**Comment by snehaljos** [comment]
Hi @devth am interested in this issue can I?

**Comment by devth** [comment]
Hi @snehaljos, I only suggested it. I'm not sure if Backstage core devs would accept a PR for it - maybe check with them first?

**Comment by Rugvip** [comment]
Yep makes sense 👍 but let's only allow it in development though. There's some prior art over here to use as inspiration: https://github.com/backstage/backstage/blob/da6e8838f3fefd3c88f6f4d826484b69e5da0bf4/plugins/rollbar-backend/src/service/router.ts#L98-L105

**Comment by devth** [comment]
I started a draft on this - just need a little guidance please! 🙏🏻 
https://github.com/backstage/backstage/pull/9525
